Source/WebGPU/WebGPU/RenderPipeline.mm

```diff
@@ -1767,7 +1764,7 @@ static uint32_t componentsForDataType(MTLDataType dataType)
 
     auto& fragment = *m_descriptor.fragment;
     for (size_t i = 0, maxTargetCount = std::max<size_t>(fragment.targetCount, descriptor.colorFormatCount); i < maxTargetCount; ++i) {
-        auto colorFormat = i < descriptor.colorFormatCount ? descriptor.colorFormats[i] : WGPUTextureFormat_Undefined;
+        auto colorFormat = i < descriptor.colorFormatCount ? span(descriptor)[i] : WGPUTextureFormat_Undefined;
```

Comment:
It seems a little odd that `span(descriptor)[i]` returns the color format, since descriptor may have two or more arrays